### ISSUE TITLE
Content generation qulaity

### ISSUE THREAD

**Issue by paramjeet2021** [kingoflolz/mesh-transformer-jax#106]
I am using this repo to train custom model. I have used approx 2-3MB of text to train my custom model and loss parameters comes around 0 after 25000 iterations so I decided to use 25000 iterations and trained my model. 
When I use this custom trained model to generate new content then the issue is generated content is not of good quality. 

Any recommendations to improve results?
Any benchmarking on the volume of the data to train a good model such that it generate quality content.

**Comment by kingoflolz** [comment]
You need at least 10x more data, ideally 100x. You should also train for significantly less as your model is overfitting.

**Comment by StellaAthena** [comment]
You’re doing too many passes over the data. Transformers are typically trained looking at each data point once or twice, but your model is going to see each data point hundreds of times.

**Comment by paramjeet2021** [comment]
Are you suggesting to reduce the number of iterations?

**Comment by StellaAthena** [comment]
> Are you suggesting to reduce the number of iterations?

Yes. I recommend setting the number of iterations to be equal to the number of tokens in the dataset, divided by the batch size, divided by the context length.

**Comment by Alexmhack** [comment]
I think this would be a good place to clear my doubts. For a 41GB `.jsonl.zst` file the `tfrecords` files size for me is exceeding 180GB. So when @kingoflolz says we need 100X more data size then do you mean `.jsonl.zst` file size should be around 200GB  - 300GB or `tfrecords` size?

And also is it unusual for `tfrecords` to be this big for just 41GB of data? because then **The Pile** `tfrecords` would have taken huge lots of space.
